src/Integration.Vsix.UnitTests/CFamily/CMake/CompilationDatabaseRequestTests.cs

```diff
@@ -0,0 +1,327 @@
+            act.Should().ThrowExactly<ArgumentException>();
+        }
+
+        [TestMethod]
+        public void WriteReqest_ValidRequest_ExpectedHeaderFooterAndSimpleProperties()
+        {
+            var dbEntry = new CompilationDatabaseEntry { File = "file.txt", Directory = "c:\\aaa", Command = "any" };
+            var context = new RequestContext("any", Mock.Of<ICFamilyRulesConfig>(), null, "d:\\preamble.txt", null);
+
+            var tokens = WriteRequest(dbEntry, context);
+
+            // Header and footer
+            tokens.First().Should().Be("SL-IN");
+            tokens.Last().Should().Be("SL-END");
+
+            // Simple properties i.e. ones that are just written as-is
+            CheckExpectedSetting(tokens, "File", "file.txt");
+            CheckExpectedSetting(tokens, "Directory", "c:\\aaa");
+            CheckExpectedSetting(tokens, "PreambleFile", "d:\\preamble.txt");
+        }
+
+        [TestMethod]
+        public void WriteReqest_WithCommand_ExpectedSettingWritten()
+        {
+            var dbEntry = new CompilationDatabaseEntry { File = "file.txt", Directory = "c:\\aaa", Command = "cmd1 cmd2" };
+
+            var tokens = WriteRequest(dbEntry, ValidContext);
+
+            CheckExpectedSetting(tokens, "Command", "cmd1 cmd2");
+            CheckSettingDoesNotExist(tokens, "Arguments");
+        }
+
+        [TestMethod]
+        public void WriteReqest_WithArguments_ExpectedSettingWritten()
+        {
+            var dbEntry = new CompilationDatabaseEntry { File = "file.txt", Directory = "c:\\aaa", Arguments = "arg1\narg2" };
+
+            var tokens = WriteRequest(dbEntry, ValidContext);
+
+            CheckExpectedSetting(tokens, "Arguments", "arg1\narg2");
+            CheckSettingDoesNotExist(tokens, "Commands");
+        }
+
+        [TestMethod]
+        [DataRow(true, "true")]
+        [DataRow(false, "false")]
+        public void WriteReqest_CreateReproducer_ExpectedSettingWritten(bool createReproducer, string expectedValue)
+        {
+            var analyzerOptions = new CFamilyAnalyzerOptions { CreateReproducer = createReproducer };
+            var context = CreateContext(analyzerOptions: analyzerOptions);
+
+            var tokens = WriteRequest(ValidDbEntry, context);
+
+            CheckExpectedSetting(tokens, "CreateReproducer", expectedValue);
+        }
+
+        [TestMethod]
+        [DataRow(true, "true")]
+        [DataRow(false, "false")]
+        public void WriteReqest_CreatePreamble_ExpectedSettingWritten(bool createPch, string expectedValue)
+        {
+            var analyzerOptions = new CFamilyAnalyzerOptions { CreatePreCompiledHeaders = createPch };
+            var context = CreateContext(analyzerOptions: analyzerOptions);
+
+            var tokens = WriteRequest(ValidDbEntry, context);
+
+            CheckExpectedSetting(tokens, "BuildPreamble", expectedValue);
+        }
+
+        [TestMethod]
+        public void WriteReqest_QualityProfile_ExpectedSettingWritten()
```

Comment:
There is a [code](https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/CFamily/CFamilyHelper.cs#L116) in the vcx that rules can be null when it's a PCH build.
We have a choice here we would we want to put that logic, if in the caller then the Request would need to be able to handle null rules.

Reply:
Good point. I'll do the same here so we don't write the setting - every millisecond counts...